### PR TITLE
Issue255

### PR DESCRIPTION
**Pull request by benlapETS**
Completed test_sr_cache.py and also added multiple comments/fixme/todo in sr_cache

### CODE REVIEW

**Comment by petersilva** [comment]
"caching for what purpose" ... yeah.. the dev that originally wrote it called it "caching" and I changed the name of the keyword to "duplicate_suppression" ... preventing *identical* (for some kinds of identical) files from being processed (either posted, or consumed for download or sending) multiple times.

discussed fairly thoroughly in the sr_subscribe man page:

https://github.com/MetPX/sarracenia/blob/master/doc/sr_subscribe.1.rst#suppress-duplicates-off-on-999-smhdw-default-off

File: sarra/sr_cache.py [hunk 1]
```diff
@@ -143,25 +156,32 @@ def check(self, key, path, part):
...
+        # FIXME meaningless debug msg
         self.logger.debug("differ")
 
         return True
 
     def check_msg(self, msg):
+        # TODO document in docstring: sr_cache may raise OSErrors this is unhandled at this point
         self.logger.debug("sr_cache check_msg")
 
+        # TODO repeated code: in sr_cache.check consider extracting to a method
```
Comment (benlapETS) [inline sarra/sr_cache.py:184]: enlever TODO

File: sarra/sr_cache.py [hunk 2]
```diff
@@ -177,7 +197,9 @@ def check_msg(self, msg):
 
         return self.check(sumstr,relpath,partstr)
 
-    def clean(self, fp = None, delpath = None):
+    def clean(self, fp=None, delpath=None):
```
Comment (benlapETS) [inline sarra/sr_cache.py:200]: change to arg persist, separate commit

File: sarra/sr_cache.py [hunk 3]
```diff
@@ -212,7 +234,7 @@ def clean(self, fp = None, delpath = None):
                 ndict[value] = t
                 self.count  += 1
 
-                if fp : fp.write("%s %f %s %s\n"%(key,t,qpath,part))
+                if fp : fp.write("%s %f %s %s\n"%(key,t,qpath,part))  # TODO need justification: no try-except for a write
```
Comment (benlapETS) [inline sarra/sr_cache.py:237]: à enlever TODO

File: sarra/sr_cache.py [hunk 4]
```diff
@@ -224,12 +246,12 @@ def close(self, unlink=False):
         try   :
                 self.fp.flush()
                 self.fp.close()
-        except: pass
+        except: pass  # TODO need justification: why are we silently excepting io error ? this is a bare except
```
Comment (benlapETS) [inline sarra/sr_cache.py:249]: log error and with except Exception

File: sarra/sr_cache.py [hunk 5]
```diff
@@ -240,19 +262,19 @@ def delete_path(self, delpath):
         # close,remove file, open new empty file
         self.fp.close()
         if os.path.exists( self.cache_file ):
-            os.unlink( self.cache_file )
+            os.unlink( self.cache_file )  # FIXME inconsistencies: to try-except or not to try-except
```
Comment (benlapETS) [inline sarra/sr_cache.py:265]: à enlever

File: sarra/sr_cache.py [hunk 6]
```diff
@@ -240,19 +262,19 @@ def delete_path(self, delpath):
...
-        self.clean(self.fp, delpath)
+        self.clean(self.fp, delpath)  # FIXME passing fp: this is an attribute why passing it as an arg ?
 
     def free(self):
         self.logger.debug("sr_cache free")
         self.cache_dict = {}
         self.count      = 0
         try   : os.unlink(self.cache_file)
-        except: pass
+        except: pass  # TODO need justification: why are we silently excepting io error ? this is a bare except
```
Comment (benlapETS) [inline sarra/sr_cache.py:277]: error msg and not BaseException

**Pull request merged**